File: NHSOnline.Backend.Worker.Mocking/Emis/Models/LinkApplicationRequest.cs
namespace NHSOnline.Backend.Worker.Mocking.Emis.Models
{
    public class LinkApplicationRequest
    {
        public string Surname { get; set; }
        public string DateOfBirth { get; set; }
        public LinkageDetails LinkageDetails { get; set; }

        public LinkApplicationRequest(string surname, string dateOfBirth, string accountId,
            string linkageKey, string nationalPracticeCode)
        {
            Surname = surname;
            DateOfBirth = dateOfBirth;
            LinkageDetails = new LinkageDetails(accountId, linkageKey, nationalPracticeCode);
        }
    }
}

File: bddtests/src/main/kotlin/mocking/emis/models/LinkApplicationRequest.kt
package mocking.emis.models

import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(name = "LinkApplicationRequest")
data class LinkApplicationRequest(
        @field:XmlElement(name = "Surname") var surname: String,
        @field:XmlElement(name = "DateOfBirth") var dateOfBirth: String) {

    @field:XmlElement(name = "LinkageDetails")
    var linkageDetails: LinkageDetails? = null

    constructor(surname: String, dateOfBirth: String,
                accountId: String, linkageKey: String,
                nationalPracticeCode: String) : this(surname, dateOfBirth) {
        linkageDetails = LinkageDetails(accountId, linkageKey, nationalPracticeCode)
    }
}